accounting/invoice/utils.py

```diff
@@ -122,8 +122,8 @@ def send_email_with_invoice(template, subject, extra_email_context=None, client_
                 invoice = models.Invoice.objects.create(
                     number=choices.InvoiceNumberingScheme.increment_value(
                         invoice.template.numbering_scheme,
-                        value=invoice.number
-                    ) if invoice else choices.InvoiceNumberingScheme.default_default(),
+                        invoice.number
+                    ) if invoice else provider.invoice_template.numbering_scheme,
```

Comment:
@UmanShahzad This will set `number` to the pattern that a provider prefers for invoice numbers (e.g. `"%Y-%m"`), but what we want is an actual number. So I think you'll need to add a `get_default_value` method to `InvoiceNumberingScheme` that works the same way as `increment_value` (calls the appropriate `default_*` method based on the numbering scheme passed in).